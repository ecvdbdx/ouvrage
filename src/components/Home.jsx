import React from 'react';

class Promo extends React.Component {
  constructor () {
    super();

    this.state = {
      chief: ''
    };
  }

  componentDidMount () {
    const studentsPromo = this.props.promo.students;
    let chiefPromo = null;
    this.props.chiefs.forEach(function (chief) {
      if (studentsPromo.includes(parseInt(chief.id, 10))) {
        chiefPromo = chief;
      }
    });
    this.setState({
      chief: chiefPromo
    });
  }

  render () {
    if (this.state.chief != null) {
      return (
        <li>
          <article>
            <header>
              <h3>
                {this.props.promo.degree}
                <span>
                  {this.props.promo.year}
                </span>
              </h3>
            </header>
            <div className="promo-content">
              <img src={this.state.chief.avatar}/>
              <div className="content-side">
                <p>
                  Délégué : {this.state.chief.first_name}
                  {this.state.chief.last_name}
                  AKA {this.state.chief.username}
                </p>
                <a href="#">Voir la classe</a>
              </div>
            </div>
          </article>
        </li>
      );
    } else {
      return (
        <li>
          <article>
            <header>
              <h3>
                {this.props.promo.degree}
                <span>
                  {this.props.promo.year}
                </span>
              </h3>
            </header>
            <div className="promo-content">
              <img src="https://fakeimg.pl/250x100/" />
              <div className="content-side">
                <p>
                  Pas de délégué
                </p>
                <a href="#">Voir la classe</a>
              </div>
            </div>
          </article>
        </li>
      );
    }
  }
}

class PromoList extends React.Component {
  render () {
    var listClass = this.props.promos.map(promo => <Promo promo={promo} key={promo.id} chiefs={this.props.chiefs}/>);

    return (<ol className="grid-3">
      {listClass}
    </ol>);
    }
}

export default class Home extends React.Component {
  render () {
    return (
      <div>
        <section>
          <header>
            <h2>L'ECV DIGITAL</h2>
          </header>
          <p>
            L’ECV Digital a été créée afin de combler les besoins du marché de l’emploi qui manque cruellement de profils web pour accompagner la transformation digitale de toutes les entreprises et de l’économie en général.
            Située à Paris, Nantes, Bordeaux et Aix-en-Provence, l’ECV Digital, l’école du numérique et du Web, forme à tous les métiers du digital.
            Issue du groupe ECV (École de Communication Visuelle), l’école se positionne comme une école généraliste du secteur internet. Son objectif est de former des profils polyvalents intégrant les compétences techniques, créatives et marketing.
          </p>
        </section>
        <section>
          <header>
            <h2>Voici la liste des classes</h2>
          </header>
          <PromoList promos={this.props.promos} chiefs={this.props.chiefs}/>
        </section>
      </div>
    );
  }
}
