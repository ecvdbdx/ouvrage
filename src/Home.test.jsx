import React from 'react';
import renderer from 'react-test-renderer';
import Home from './components/Home';
import promos from '../mocks/classes.json';

it('renders correctly', () => {
  const tree = renderer.create(<Home promos={promos} chiefs={[[]]}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
