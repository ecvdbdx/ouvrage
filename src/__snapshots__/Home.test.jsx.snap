// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <section>
    <header>
      <h2>
        L'ECV DIGITAL
      </h2>
    </header>
    <p>
      L’ECV Digital a été créée afin de combler les besoins du marché de l’emploi qui manque cruellement de profils web pour accompagner la transformation digitale de toutes les entreprises et de l’économie en général. Située à Paris, Nantes, Bordeaux et Aix-en-Provence, l’ECV Digital, l’école du numérique et du Web, forme à tous les métiers du digital. Issue du groupe ECV (École de Communication Visuelle), l’école se positionne comme une école généraliste du secteur internet. Son objectif est de former des profils polyvalents intégrant les compétences techniques, créatives et marketing.
    </p>
  </section>
  <section>
    <header>
      <h2>
        Voici la liste des classes
      </h2>
    </header>
    <ol
      className="grid-3"
    >
      <li>
        <article>
          <header>
            <h3>
              m1
              <span>
                2017/2018
              </span>
            </h3>
          </header>
          <div
            className="promo-content"
          >
            <img
              src="https://fakeimg.pl/250x100/"
            />
            <div
              className="content-side"
            >
              <p>
                Pas de délégué
              </p>
              <a
                href="#"
              >
                Voir la classe
              </a>
            </div>
          </div>
        </article>
      </li>
      <li>
        <article>
          <header>
            <h3>
              m2
              <span>
                2017/2018
              </span>
            </h3>
          </header>
          <div
            className="promo-content"
          >
            <img
              src="https://fakeimg.pl/250x100/"
            />
            <div
              className="content-side"
            >
              <p>
                Pas de délégué
              </p>
              <a
                href="#"
              >
                Voir la classe
              </a>
            </div>
          </div>
        </article>
      </li>
      <li>
        <article>
          <header>
            <h3>
              webD1
              <span>
                2017/2018
              </span>
            </h3>
          </header>
          <div
            className="promo-content"
          >
            <img
              src="https://fakeimg.pl/250x100/"
            />
            <div
              className="content-side"
            >
              <p>
                Pas de délégué
              </p>
              <a
                href="#"
              >
                Voir la classe
              </a>
            </div>
          </div>
        </article>
      </li>
      <li>
        <article>
          <header>
            <h3>
              webD2
              <span>
                2017/2018
              </span>
            </h3>
          </header>
          <div
            className="promo-content"
          >
            <img
              src="https://fakeimg.pl/250x100/"
            />
            <div
              className="content-side"
            >
              <p>
                Pas de délégué
              </p>
              <a
                href="#"
              >
                Voir la classe
              </a>
            </div>
          </div>
        </article>
      </li>
    </ol>
  </section>
</div>
`;
